src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/FileWatching/LspFileChangeWatcher.cs

```diff
@@ -76,7 +76,7 @@ public FileChangeContext(ImmutableArray<WatchedDirectory> watchedDirectories, Ls
                 {
                     GlobPattern = new RelativePattern
                     {
-                        BaseUri = d.Path,
+                        BaseUri = new Uri(d.Path, UriKind.Absolute),
```

Comment:
We do have a helper ProtocolConversions.GetUriFromFilePath if you want to call that. It's not a hugely magic helper but at least would mean any bug fixes would apply there too.